tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -247,6 +248,23 @@ def with_poetry(context: PipelineContext) -> Container:
     return poetry_with_cache
 
 
+def with_pipx(context: PipelineContext) -> Container:
+    """Installs pipx in a python environment.
+
+    Args:
+        context (PipelineContext): The current test context, providing the repository directory from which the ci_credentials sources will be pulled.
+
+    Returns:
+        Container: A python environment with pipx installed.
+    """
+    install_pipx_package_cmd = ["python", "-m", "pip", "install", "pipx"]
+
+    python_with_poetry: Container = with_poetry(context)
+    python_with_pipx = python_with_poetry.with_exec(install_pipx_package_cmd).with_env_variable("PIPX_BIN_DIR", "/usr/local/bin")
```

Comment:
I don't really know why we use `pipx` and how related it is to `poetry`, but if pipx does not depend on poetry I think we should use `with_python_base` instead of `with_poetry` and eventually compose the two in an outer layer / environment: 
```python
python_base = with_python_base(context)
poetry_container = with_poetry(python_base)
pipx_and_poetry_container = with_pipx(poetry_container)
```